src/lib.rs

```diff
@@ -264,6 +265,31 @@ impl BlockStore {
         })
     }
 
+    pub fn open_test(path: impl AsRef<Path>, mut config: Config) -> anyhow::Result<Self> {
+        let mut conn = Connection::open_in_memory()?;
+        debug!(
+            "Restoring in memory database from {}",
+            path.as_ref().display()
+        );
+        conn.restore(
+            DatabaseName::Main,
+            path,
+            Some(|p: rusqlite::backup::Progress| {
+                let percent = (p.pagecount - p.remaining) * 100 / p.pagecount;
+                if percent % 10 == 0 {
```

Comment:
Not really the end of the world, but this does not guarantee that you will see any progress. You might get numbers like 9, 31, 45, ...

Not sure if we need progress at all - might be enough to log start and end.

Reply:
It's a nice touch, and some non-scientific testing showed sufficient granularity